src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -806,22 +707,72 @@ class WorkspaceService(userInfo: UserInfo, dataSource: DataSource, workspaceDAO:
         }
       }
     }
+
+  // helper methods
+
+  private def noSuchWorkspaceMessage( workspaceName: WorkspaceName ) = s"${workspaceName} does not exist"
+
+  private def withWorkspaceContextAndPermissions(workspaceName: WorkspaceName, accessLevel: GCSAccessLevel, txn: RawlsTransaction)(op: (WorkspaceContext) => PerRequestMessage): PerRequestMessage = {
+    withWorkspaceContext(workspaceName, txn) { workspaceContext =>
+      requireAccess(workspaceName, workspaceContext.bucketName, accessLevel, txn) { op(workspaceContext) }
+    }
   }
 
-  def requireAccess(requiredLevel: GCSAccessLevel, workspace: Workspace, txn: RawlsTransaction)(codeBlock: => PerRequestMessage): PerRequestMessage = {
-    val acls = JsonParser(gcsDAO.getACL(userInfo.userId, workspace.bucketName)).convertTo[BucketAccessControls]
-    if (acls.maximumAccessLevel >= requiredLevel)
-      codeBlock
-    else
-      RequestComplete(http.StatusCodes.NotFound, noSuchWorkspaceMessage(workspace.toWorkspaceName))
+  private def withWorkspaceContext(workspaceName: WorkspaceName, txn: RawlsTransaction)(op: (WorkspaceContext) => PerRequestMessage) = {
+    workspaceDAO.loadContext(workspaceName, txn) match {
+      case None => RequestComplete(http.StatusCodes.NotFound, noSuchWorkspaceMessage(workspaceName))
+      case Some(workspaceContext) => op(workspaceContext)
+    }
   }
 
-  def requireAccess(requiredLevel: GCSAccessLevel, workspaceName: WorkspaceName, txn: RawlsTransaction)(codeBlock: => PerRequestMessage): PerRequestMessage = {
-    withWorkspace(workspaceName, txn) { workspace =>
-      requireAccess(requiredLevel, workspace, txn)(codeBlock)
+  private def withWorkspace(workspaceName: WorkspaceName, txn: RawlsTransaction)(op: (Workspace) => PerRequestMessage) = {
+    workspaceDAO.load(workspaceName, txn) match {
+      case None => RequestComplete(http.StatusCodes.NotFound, noSuchWorkspaceMessage(workspaceName))
+      case Some(workspace) => op(workspace)
     }
   }
 
+  private def withEntity(workspace: WorkspaceContext, entityType: String, entityName: String, txn: RawlsTransaction)(op: (Entity) => PerRequestMessage): PerRequestMessage = {
```

Comment:
any WorkspaceContext variable i think we should call workspaceContext to be clear